### PR TITLE
Interpolate any column

### PR DESCRIPTION
**Pull request by MarionJHolloway**
# Description
Add a parameter to interpolation function to allow for it to be run on any column

# How to test
Unit tests passing
Run in branch: https://eu-west-2.console.aws.amazon.com/states/home?region=eu-west-2#/v2/executions/details/arn:aws:states:eu-west-2:344210435447:execution:interpolate-any-column-Ind-CQC-Filled-Post-Estimates-Pipeline:42f994ce-6d77-463a-b0b2-5ea532d66db6

# Developer checklist
- [x] Unit test
- [x] Linked to Trello ticket: https://trello.com/c/xZHhnArq/738-add-columntointerpolate-to-interpolation-function-must
- [x] Documentation up to date

### CODE REVIEW

File: utils/estimate_filled_posts/models/interpolation.py [hunk 1]
```diff
@@ -86,35 +86,39 @@ def leftouter_join_on_locationid_and_unix_time(
...
             F.col(IndCqc.unix_time),
         ).otherwise(F.lit(None)),
     )
     return df
 
 
-def interpolate_values_for_all_dates(df: DataFrame) -> DataFrame:
-    df = input_previous_and_next_values_into_df(df)
-    df = calculate_interpolated_values_in_new_column(df, IndCqc.interpolation_model)
+def interpolate_values_for_all_dates(
```
Comment (garypolzin) [inline utils/estimate_filled_posts/models/interpolation.py:102]: add docstring

File: utils/estimate_filled_posts/models/interpolation.py [hunk 2]
```diff
@@ -86,35 +86,39 @@ def leftouter_join_on_locationid_and_unix_time(
...
+        df, IndCqc.interpolation_model, column_to_interpolate
+    )
     return df
 
 
-def input_previous_and_next_values_into_df(df: DataFrame) -> DataFrame:
-    df = get_previous_value_in_column(
-        df, IndCqc.ascwds_filled_posts_dedup_clean, IndCqc.previous_filled_posts
-    )
+def input_previous_and_next_values_into_df(
```
Comment (garypolzin) [inline utils/estimate_filled_posts/models/interpolation.py:112]: add docstring

File: utils/estimate_filled_posts/models/interpolation.py [hunk 3]
```diff
@@ -182,18 +186,18 @@ def calculate_interpolated_values_in_new_column(
 
 def interpolate_values(
```
Comment (garypolzin) [inline utils/estimate_filled_posts/models/interpolation.py:187]: add docstring

File: utils/estimate_filled_posts/models/interpolation.py [hunk 4]
```diff
@@ -8,34 +8,34 @@
...
     return df
 
 
-def filter_to_locations_with_a_known_filled_posts(df: DataFrame) -> DataFrame:
-    df = df.select(
-        IndCqc.location_id, IndCqc.unix_time, IndCqc.ascwds_filled_posts_dedup_clean
-    )
+def filter_to_locations_with_a_known_value(
+    df: DataFrame, column_to_interpolate: str
+) -> DataFrame:
```
[the remark this answers is not in the document]
Comment (MarionJHolloway) [reply utils/estimate_filled_posts/models/interpolation.py:35]: Will do doc strings in next PR.

**Pull request merged**